packages/@o3r/core/schematics/sdk/index.ts

```diff
@@ -39,15 +39,29 @@ export function generateSdk(options: NgGenerateSdkSchema): Rule {
       renameTemplateFiles()
     ]), MergeStrategy.Overwrite);
 
+    const packageManager = getPackageManager({ workspaceConfig });
+    const yarnrcPath = '.yarnrc.yml';
+    const yarnrcBeforeSdkGeneration = tree.exists(yarnrcPath) ? tree.readText(yarnrcPath) : '';
+
     return chain([
       externalSchematic<NgGenerateTypescriptSDKShellSchematicsSchema>('@ama-sdk/schematics', 'typescript-shell', {
         ...options,
         package: projectName,
         name: scope,
         directory: targetPath,
-        packageManager: getPackageManager({workspaceConfig}),
+        packageManager,
         skipInstall: !!options.specPath || options.skipInstall
       }),
+      packageManager === 'yarn' ? (t) => {
+        if (yarnrcBeforeSdkGeneration) {
+          // discard changes done by sdk shell generator standalone on yarnrc
+          t.overwrite(yarnrcPath, yarnrcBeforeSdkGeneration);
```

Comment:
Why filter out the yarnrc from the shell instead of not adding it in the first place (via a param for example)?
Wouldn't you also get the same problem with the npmrc file?

Reply:
The 'yarnrc' is added by the shell generator. I did not want to do it with a parameter, just to detect if we are running the sdk generator in standlone repo or in a monorepo. We already have the case to remove some generated files (generated by shell generator for standlone repo) and we can do that because we know we are in a generator which runs in a monorepo. Like that we don;t polute the shell generator with another param.

`Wouldn't you also get the same problem with the npmrc file?`
We'll not have the same problem because in the shell generator we check if we use yarn as package manager.